File: 3_private_encaptulation.py
# https://www.geeksforgeeks.org/encapsulation-in-cpp/
class Encapsulation:
    def __init__(self):
        self.__a = None #private number
        self.__b = None #private number

    def sum_two_numbers(self,x,y):
        self.__a=x
        self.__b=y
        return x+y
    def get_numbers(self):
        print(f"a->{self.__a}")
        print(f"b->{self.__b}")
    

En = Encapsulation()
En.sum_two_numbers(4,5)
En.get_numbers()
# print(En.__a) AttributeError: 'Encapsulation' object has no attribute '__a'
# print(En.__b) AttributeError: 'Encapsulation' object has no attribute '__a'

File: 3_private_encaptulation.cpp
// https://www.geeksforgeeks.org/encapsulation-in-cpp/
#include <bits/stdc++.h>
#include<iostream>
#include <string>
using namespace std;

class Encapsulation{
    private:
        int a,b; 
        // You can not access outside this class.
    public:
        int sum_two_numbers(int x, int y){
            this->a=x;
            this->b=y;
            return x+y;
        }
        void get_numbers(){
            cout<<"a->"<<this->a<<endl;
            cout<<"b->"<<this->b<<endl;
        }
};

int main(){
    Encapsulation En;
    int num = En.sum_two_numbers(4,5);
    cout<<num<<endl;
    //cout<<En.a<<endl; //member "Encapsulation::a" (declared at line 9) is inaccessibleC/C++(265)
    //cout<<En.b<<endl; //member "Encapsulation::a" (declared at line 9) is inaccessibleC/C++(265)
    En.get_numbers();
}